File: apps/web/src/types/book.ts
/**
 * 교재 관련 타입 정의
 */
export enum ParseStatus {
  PENDING = "PENDING",
  PROCESSING = "PROCESSING",
  DONE = "DONE",
  FAILED = "FAILED",
}

export enum Subject {
  KOREAN = "KOREAN",
  ENGLISH = "ENGLISH",
  MATH = "MATH",
}

export interface Book {
  book_id: string;
  title: string;
  subject: Subject;
  year?: number;
  parse_status: ParseStatus;
  lesson_count?: number;
}

export interface BookCreate {
  title: string;
  subject: Subject;
  year?: number;
}

export interface BookParseStatus {
  book_id: string;
  status: ParseStatus;
  progress: number;
}

File: backend/app/schemas/book.py
"""
교재 관련 Pydantic 스키마
"""
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from app.infrastructure.database.models import ParseStatus, Subject


class BookCreate(BaseModel):
    title: str
    subject: Subject
    year: Optional[int] = None


class BookResponse(BaseModel):
    book_id: str
    title: str
    subject: Subject
    year: Optional[int]
    parse_status: ParseStatus
    lesson_count: Optional[int] = 0
    
    class Config:
        from_attributes = True


class BookParseStatusResponse(BaseModel):
    book_id: str
    status: ParseStatus
    progress: int  # 0-100
    current_page: Optional[int] = 0
    total_pages: Optional[int] = 0
    message: Optional[str] = None